### DISCUSSION TITLE
Converting u_0 and t_E

### DISCUSSION THREAD

**Discussion by adammcmaster** [rpoleski/MulensModel#47] [Q&A] [answered]
I'm working on a project searching for BH self lensing events in an archive of photometric data, and I'm looking into using MulensModel to fit the light curves of potential lensing events. As well as the real observations we also have simulations that we're using for verification, so the first thing I want to do is to fit our simulated light curves and check that I can recover the original simulation parameters.

I've found it really easy to produce a fit with MulensModel (so good work on building something that's so straightforward to use) and the fits look really good when I plot them, so I'm convinced that the fitted _u_0_ and _t_E_ are correct. The problem I have is that those aren't the same parameters that were used in the simulations!

Details of the simulations are published in [this paper](https://arxiv.org/abs/2104.12666) but basically they're based on the [parameters given here](https://universeathome.pl/universe/self_lensing.php): the relevant ones being _b_ and _tau_eff_. 

Now, I'm pretty sure that for a self-lensing event _b_ and _u_0_ should be more or less the same, since the distance between the source and the lens is so small, but they're not even close. And I guess _t_E_ and _tau_eff_ aren't the same but I think they should be related to each other. In one example I have _b=0.2093_ vs _u_0=1.4_ and _tau_eff=19.24_ vs _t_E=5.49_.

Like I said before, the fits look really good so I don't think the fitted parameters are wrong. I think I probably just need to do a unit conversion or something, but I've tried everything I can think of and reached a dead end! Can anyone point me in the right direction?

**Comment by rpoleski** [comment, marked as answer]
Good to hear that MM is useful! Please let me note that early-on we decided to not include binary self-lensing in MM. The parameterization of self-lensing is completely different :( I think you can use MM only for calculating 1 and 2 from the linked paper. Then you should write a separate code that calculates magnification as a function of time and parameterized by b, t_eff etc. The fact that you can fit using t_E, u_0 etc. parameterization from MM is a coincidence - the fitted signal has relatively simple shape. I suggest [Kruse & Agol (2014)](https://ui.adsabs.harvard.edu/abs/2014Sci...344..275K/abstract) for equations of A(t) for a self-lensing system. If you have a code that uses MM as part of such calculations, then I would be happy to add it as one of the examples to the repo.

**Reply by adammcmaster** [reply]
Ah! That's a shame, but it shouldn't be a problem to do it that way. Thanks for getting back to me so quickly!
